NYPL-discovery/discovery-front-end
# User EDD Form Session Storage

**What's this do?**
A method to which we can save the user form input and reuse the inputs if the users login session expires is to set the value on the browsers session state.

This PR processes the users form session state in three ways:
1. It captures the data input and sets the values, on update, in the session state.
2. It uses the form session input if it exists and after checking for passed in forms props,
3. It removes the values when the form is submitted.
    - Since the submittance of the form inputs do not return a value
    - and the page is redirected on success or failure
    - we can simply remove saved session item

**Why are we doing this? (w/ JIRA link if applicable)**
When a user is on the EDD submittal form and their login session is forced to be revalidated, capture and set any form inputs added by the user. This is a feature which provides convenience to the user if they fail to submit their request in the time allotted.

**Do these changes have automated tests?**
Yes - Located in ElectronicDeliver.test.js

**How should this be QAed?**
1. On Initial render, ensure no values are present in form input fields except for patrons email.
2. After input entry, ensure a local storage object is available (called `formstate`) with a property of the `itemId`
3. On page refresh or with same `itemId` the form input fields are prefilled with previously entered values
4. On submittal of the form, ensure local storage has been cleared
5. On re-entry of the EDD request after submittal, ensure clean form state

**Dependencies for merging? Releasing to production?**
None

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Unknown
